Engine/Program.cs

```diff
@@ -0,0 +1,239 @@
+﻿// See https://aka.ms/new-console-template for more information
+
+using Engine.CliHelpers;
+using Engine.Helpers;
+
+namespace Engine;
+
+public static class Program
+{
+    public enum BotDifficulty
+    {
+        Easy,
+        Medium,
+        Hard,
+        Impossible
+    }
+
+    public static Random random = new();
+
+    public static Dictionary<BotDifficulty, BotData> Bots = new()
+    {
+        {
+            BotDifficulty.Easy,
+            new BotData
+            {
+                Name = "Limping Liam", CustomIntroMessage = "He can't jump far", CustomLoseMessage = "Oh no",
+                CustomWinMessage = "Easy", QuickestResponseTimeMs = 3000,
+                SlowestResponseTimeMs = 5000
+            }
+        },
+        {
+            BotDifficulty.Medium,
+            new BotData
+            {
+                Name = "Harrowing Hayden", CustomIntroMessage = "He's a bit of a trickster so watch out",
+                CustomLoseMessage = "Damn, he's tricky", CustomWinMessage = "Down goes the trickster",
+                QuickestResponseTimeMs = 2000, SlowestResponseTimeMs = 4000
+            }
+        },
+        {
+            BotDifficulty.Hard,
+            new BotData
+            {
+                Name = "Masterful Mikaela", CustomIntroMessage = "She can't be trusted",
+                CustomLoseMessage = "Oof, rough one", CustomWinMessage = "Down falls Mikaela and her wicked ways",
+                QuickestResponseTimeMs = 1000,
+                SlowestResponseTimeMs = 2000
+            }
+        },
+        {
+            BotDifficulty.Impossible,
+            new BotData
+            {
+                Name = "Chaotic Kate", CustomIntroMessage = "rip lol", CustomLoseMessage = "No chance",
+                CustomWinMessage = "No one will ever see this message so it doesn't matter",
+                QuickestResponseTimeMs = 500,
+                SlowestResponseTimeMs = 1500
+            }
+        }
+    };
+
+    public static BotData Bot;
+
+    public static GameState gameState { get; set; }
+
+    public static void Main(string[] args)
+    {
+        PlayGame();
+
+        void PlayGame(bool skipIntro = false)
+        {
+            // Difficulty
+            BotDifficulty botDifficulty = CliGameUtils.GameIntro(skipIntro);
+            Bot = Bots[botDifficulty];
+            (int min, int max) botSpeedMs = (Bot.QuickestResponseTimeMs, Bot.SlowestResponseTimeMs);
+
+            gameState = GameEngine.NewGame(new List<string> {Bot.Name, "You"});
+            UpdateMessage("Game started!");
+
+            // Main game loop
+            while (GameEngine.TryGetWinner(gameState).Failure)
+            {
+                HandleUserInput(ReadlineWithBot());
+            }
+
+            // End game
+            Player winner = GameEngine.TryGetWinner(gameState).Data;
+            bool winnerIsPlayer = gameState.Players.IndexOf(winner) == 1;
+
+            Console.WriteLine("------ Game over ----");
```

Comment:
I think you should continue using CLIUtils or another class to be responsible for this sort of console output and move it out of `Program`. Another thing I try to think about is have things of similar granularity in a method. E.g. If you already have calls like `CliGameUtils.GameIntro()` and `HandleUserInput` etc that are at a quite a high level then I would try to avoid mixing low level logic like `WriteLine` etc in with it to improve readability